triton/kubernetes/AWS/FIL/README.md

```diff
@@ -0,0 +1,215 @@
+## Deploying a Triton service that uses a custom plugin for the RAPIDS cuML Forest Inference Library (FIL).
+
+**NOTE:** For steps to setup a horizontally autoscalable Triton Service in EKS refer to the instructions in [Detailed_HPA_Setup](./Detailed_HPA_Setup.md).
+
+### Overview
+
+This example will illustrate the workflow to deploy a Triton inference service, with a [non-trival custom backend](https://github.com/triton-inference-server/fil_backend) for accelerated forest model inference on [Amazon Elastic Kubernetes Service (EKS)](https://aws.amazon.com/eks/?whats-new-cards.sort-by=item.additionalFields.postDateTime&whats-new-cards.sort-order=desc&eks-blogs.sort-by=item.additionalFields.createdDate&eks-blogs.sort-order=desc). 
+
+It is assumed that you have an existing account with sufficient compute and GPU quota, a correctly configured `kubectl` utility, and a properly configured and accessible AWS S3 bucket.
+
+**NOTE:** When referring to configuration parameters or elements that are `specific to your environment`, they will be represented as linux-style environment variables ex. ${YOUR_CONFIG_PARAM}.
+
+ 
+Specific parameters used here include:
+
+- REGION_NAME= < your preferred location >
+- EKS_CLUSTER_NAME= < your cluster name >
+- INSTANCE_TYPE= g4dn.12xlarge # or any other VM size with GPUs.
+- AWS_ACCOUNT_ID= < select your ACCOUNT ID >
+- REPOSITORY_NAME= titon_fil_backend # < name of the custom image we are going to build for triton backend, you can change this to your preference >
+- S3_BUCKET_PATH= < s3 bucket name (existing or to be created) to host the models >
+
+  
+---
+
+### Step 0: Pre-requisites
+
+- AWS account with at least the following permissions mentioned in https://eksctl.io/usage/minimum-iam-policies/ along with permissions to create, pull and push to AWS ECR repositories.
+- System Softwares
+    -  `docker`
+    -  `aws-cli` , `eksctl`
+    -  `helm` , `kubectl` , `istioctl`
+- Python Libraries
+```shell
+pip install nvidia-pyindex
+pip install tritonclient[all]
+```
+- Login credentials for aws-cli and kubectl configured.
+
+**Note:** The rest of the demo uses `aws-cli`  [version 2](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html). Hence some commands will be a little different from `aws-cli` version 1. We will also be using `helm` 3 that does not need `tiller`.
+
+  
+
+### Step 1: Obtain the Triton FIL plugin, build the triton host container, and push to ECR
+
+Note: as of this writing, the [FIL backend plugin](https://github.com/triton-inference-server/fil_backend) is supported in the official [Triton Inference Server](https://ngc.nvidia.com/catalog/containers/nvidia:tritonserver) image officially available in the [NVIDIA NGC Catalog](https://ngc.nvidia.com/catalog).
+
+However, here, we will show how to create a custom image with FIL backend.
```

Comment:
Is there a particular reason we go through this step in this example as opposed to just taking 21.06.1 off NGC?

Reply:
Yah I wrote most of these while the NGC image bug was being fixed 😉  I'll add the NGC image as the preferable approach and move the custom image build for informational purposes in a subsection. Someone following this can skip the custom step if they want to.